### PR TITLE
cleanup for #1848

### PR DESCRIPTION
**Pull request by cdietrich**
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/serializer/SerializerFragment2.xtend [hunk 1]
```diff
@@ -475,7 +474,6 @@ class SerializerFragment2 extends AbstractStubGeneratingFragment {
 				�IF partitions.size > 1�
 					�FOR partition : partitions.indexed�
 						private void init�partition.key�(�IGrammarAccess� access) {
```
Comment (cdietrich) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/serializer/SerializerFragment2.xtend:476]: should we make the initx method empty then @szarnekow @lukasf98

**Comment by cdietrich** [comment]
@lukasf98 can you please check if updated patch still works for you

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/serializer/SerializerFragment2.xtend [hunk 2]
```diff
@@ -461,8 +460,9 @@ class SerializerFragment2 extends AbstractStubGeneratingFragment {
 				@�Inject�
 				protected void init(�IGrammarAccess� access) {
 					�IF partitions.size > 1�
+						grammarAccess = (�grammar.grammarAccess�) access;
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/serializer/SerializerFragment2.xtend:463]: Nit: Can be moved above the if (the else branch needs the same code)

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/serializer/SerializerFragment2.xtend (hunk 1 above)
Comment (szarnekow) [reply org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/serializer/SerializerFragment2.xtend]: empty in the sense of `no args` ? Yes.

Comment (cdietrich) [reply org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/serializer/SerializerFragment2.xtend]: yes. 

so

init1()
init2()
init3()

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/serializer/SerializerFragment2.xtend (hunk 2 above)
Comment (cdietrich) [reply org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/serializer/SerializerFragment2.xtend]: done

**Comment by lukasf98** [comment]
> @lukasf98 can you please check if updated patch still works for you

Still works for me. Thanks a lot for the fixes.

**Review by szarnekow — APPROVED** [review]

**Pull request merged**